lib/kb_ObjectInfo/CreateFeatureLists_Report.py

```diff
@@ -12,136 +12,54 @@ def __init__(self, config):
             cat2group[namespace] = dict()
-#            domfam2cat[namespace] = dict()
-#            cat2domfams[namespace] = dict()
-#            domfam2group[namespace]  = dict()
-#            domfam2name[namespace]  = dict()
             
-            # get high-level cats
-            tigrrole_id2cat = dict()
-            with open(domain_cat_names_path[namespace], 'r') as dom_cat_handle:
-                for line in dom_cat_handle.readlines():
-                    line = line.strip()
-
-                    if namespace == 'COG':
-                        [cat, cat_group, cat_name] = line.split("\t")[0:3]
-                        if  cat not in cats:
-                            cats.append(cat)
-                        cat2name[namespace][cat] = cat_name
-                        cat2group[namespace][cat] = cat_group
-
-                    elif namespace == 'PF':
-                        [cat, cat_name] = line.split("\t")[0:2]
-                        if cat not in cats:
-                            cats.append(cat)
-                        cat2name[namespace][cat] = cat_name
-                        cat2group[namespace][cat] = 'NA'
-
-                    elif namespace == 'TIGR':
-                        if line.startswith('!'):
-                            continue
-                        [cat, cat_id, cat_group, cat_name_plus_go_terms] = line.split("\t")[0:4]
-                        tigrrole_id2cat[cat_id] = cat
-                        if cat not in cats:
-                            cats.append(cat)
-                        cat_name = re.sub(' *\> GO:.*$', '', cat_name_plus_go_terms)
-                        cat2name[namespace][cat] = cat_name
-                        cat2group[namespace][cat] = cat_group
-
-                    elif namespace == 'SEED':
-                        #[cat_group, cat_subgroup, cat, domfam] = line.split("\t")[0:4]
-                        [cat_group, cat] = line.split("\t")[0:2]
-                        if cat not in cats:
-                            cats.append(cat)
-                        #cat_disp = re.sub('_', ' ', cat)
-                        #cat2name[namespace][cat] = cat_disp
-
-            # get domfam to cat map, and vice versa
-            with open(domain_to_cat_map_path[namespace], 'r') as dom2cat_map_handle:
-                for line in dom2cat_map_handle.readlines():
-                    line = line.strip()
-
-                    if namespace == 'COG':
-                        [domfam, cat_str, dom_name] = line.split("\t")[0:3]
-                        cat = cat_str[0]  # only use first cat
+        # get high-level cats
+        with open(os.path.join(domain_desc_basepath, 'SEED_funcat.txt'), 'r') as dom_cat_handle:
+            for line in dom_cat_handle.readlines():
+                line = line.strip()
+                [cat_group, cat] = line.split("\t")[0:2]
+                if cat not in cats:
+                    cats.append(cat)
+        dom_cat_handle.close()
```

Comment:
close() should be omitted as the "with open()" will automatically close the file.